src/holofuel/components/layout/PrimaryLayout/PrimaryLayout.js

```diff
@@ -24,13 +27,36 @@ export function PrimaryLayout ({
   const { loading: holofuelUserLoading, data: { holofuelUser = {} } = {} } = useQuery(HolofuelUserQuery)
   const { loading: ledgerLoading, data: { holofuelLedger: { balance: holofuelBalance } = {} } = {} } = useQuery(HolofuelLedgerQuery, { fetchPolicy: 'cache-and-network' })
 
+  const [hasCalledReroute, setHasCalledReroute] = useState(false)
+
   const inboxCount = actionableTransactions.filter(actionableTx => actionableTx.status !== STATUS.canceled && !((actionableTx.status === STATUS.declined) && (actionableTx.type === TYPE.request))).length
 
   const isWide = useContext(ScreenWidthContext)
   const [isMenuOpen, setMenuOpen] = useState(false)
   const hamburgerClick = () => setMenuOpen(!isMenuOpen)
   const handleMenuClose = () => setMenuOpen(false)
 
+  const { push } = useHistory()
+  // eslint-disable-next-line
+  const goToInbox = useCallback(() => {
+    if (!hasCalledReroute) {
+      push(INBOX_PATH)
+      setHasCalledReroute(true)
+    }
+  })
```

Comment:
I feel vary wary of this pattern of setting state to tell useEffect whether to work or not. I understand why you're doing it here but I think there's a better way. Just not sure what yet 😆 

I'm going to do some research today.

This is not blocking but I consider it code debt (I used the same in a recent PR, and marked it with a comment as temporary)
Can you add a TODO comment, something like "TODO: research a better way of doing this"

Reply:
Good feedback and thoughts.  I'd be happy to hear results from your recent research as well as collective consider a cleaner approach. :100: